packages/grid/_modules_/grid/components/panel/filterPanel/GridFilterInputMultipleSingleSelect.tsx

```diff
@@ -0,0 +1,148 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import Autocomplete, { AutocompleteProps, createFilterOptions } from '@mui/material/Autocomplete';
+import Chip from '@mui/material/Chip';
+
+import TextField from '@mui/material/TextField';
+import { unstable_useId as useId } from '@mui/material/utils';
+import { GridFilterItem } from '../../../models/gridFilterItem';
+
+export type GridFilterInputMultipleSingleSelectProps = {
+  item: GridFilterItem;
+  applyValue: (value: GridFilterItem) => void;
+  // Is any because if typed as GridApiRef a dep cycle occurs. Same happens if ApiContext is used.
+  apiRef: any;
+  focusElementRef?: React.Ref<any>;
+  type?: 'singleSelect';
+} & Omit<AutocompleteProps<any[], true, false, true>, 'options' | 'renderInput'>;
+
+const getSingleSelectOptionFormatter =
+  ({ valueFormatter, field }, api) =>
+  (option) => {
+    if (typeof option === 'object') {
+      return option.label;
+    }
+    return valueFormatter && option !== '' ? valueFormatter({ value: option, field, api }) : option;
+  };
+
+const filter = createFilterOptions<any>();
+
+function GridFilterInputMultipleSingleSelect(props: GridFilterInputMultipleSingleSelectProps) {
+  const { item, applyValue, type, apiRef, focusElementRef, ...other } = props;
+  const [filterValueState, setFilterValueState] = React.useState(item.value || []);
+  const id = useId();
+
+  const currentColumn = item.columnField ? apiRef.current.getColumn(item.columnField) : null;
+  const currentValueOptions = React.useMemo(() => {
+    return typeof currentColumn.valueOptions === 'function'
+      ? currentColumn.valueOptions({ field: currentColumn.field })
+      : currentColumn.valueOptions;
+  }, [currentColumn]);
+
+  const filterValueOptionFormater = getSingleSelectOptionFormatter(
+    apiRef.current.getColumn(item.columnField),
+    apiRef.current,
+  );
+
+  const filterValueParser = React.useCallback(
+    (value) => String(typeof value === 'object' ? value.value : value),
+    [],
+  );
+
+  React.useEffect(() => {
+    let itemValue;
+
+    if (currentValueOptions !== undefined) {
+      // sanitize if valueOptions are provided
+      const parsedCurrentValueOptions = currentValueOptions.map(filterValueParser);
+      itemValue = item.value
+        .map(filterValueParser)
+        .filter((element) => parsedCurrentValueOptions.includes(element));
+      if (itemValue !== item.value) {
```

Comment:
You're comparing the reference, but the references will never be the same because `Array.filter` outputs a new array. This causes a infinite loop when I try to filter for a value. Either use `isDeepEqual` or compare the number of non-existing options found.